aten/src/THNN/generic/MultiMarginCriterion.c

```diff
@@ -159,9 +195,30 @@ void THNN_(MultiMarginCriterion_updateGradInput)(
     input_data += dim;
     gradInput_data += dim;
   }
+  gradInput_data = THTensor_(data)(gradInput);
+
```

Comment:
I'm actually somewhat concerned about the code above, which you didn't change. In the `d == target_idx` case, is `gradInput_data[d]` already zero?

Reply:
It shouldn't matter because the value for that case is assigned after the inner loop, at line 193.

Reply:
Makes sense. Thanks for explanation.